server/src/tests/ws.rs

```diff
@@ -673,46 +671,36 @@ async fn ws_server_backpressure_works() {
 			"n",
 			"unsubscribe_with_backpressure_aggregation",
 			move |_, pending, mut backpressure_tx| async move {
-				let mut sink = pending.accept().await?;
+				let sink = pending.accept().await?;
 				let n = sink.build_message(&1).unwrap();
 				let bp = sink.build_message(&2).unwrap();
 
-				let mut buf = VecDeque::new();
-				buf.push_back(n.clone());
-				let mut sent = 0;
-
-				while let Some(next) = buf.pop_front() {
-					if sent > 20 {
-						break;
-					}
-
-					// just try "buffer" 20 items to avoid OOM.
-					if buf.len() > 20 {
-						sink.send(next).await?;
-						sent += 1;
-					} else {
-						match sink.try_send(next) {
-							Err(TrySendError::Closed(_)) => {
-								println!("user closed");
-								break;
-							}
-							Err(TrySendError::Full(unsent)) => {
-								// send a message to the testing code to indicate that
-								// the buffer was exceeded.
-								let b_tx = std::sync::Arc::make_mut(&mut backpressure_tx);
-								let _ = b_tx.send(()).await;
-
-								buf.push_back(unsent);
-								buf.push_back(bp.clone());
-							}
-							Ok(()) => {
-								sent += 1;
-								buf.push_back(n.clone());
-							}
-						}
+				let mut msg = n.clone();
+
+				loop {
+					tokio::select! {
+						biased;
+						_ = sink.closed() => {
+							// User closed connection.
+							break;
+						},
+						res = sink.send_timeout(msg.clone(), std::time::Duration::from_millis(100)) => {
```

Comment:
I'm just wondering; do we need the `send_timeout` stuff rather than using a combinator like https://docs.rs/tokio/latest/tokio/time/fn.timeout.html?